cli/js/lib.deno.ns.d.ts

```diff
@@ -1569,6 +1574,15 @@ declare namespace Deno {
     transport?: Transport;
   }
 
+  export interface UnixListenOptions {
+    /** A Path to the Unix Socket. */
+    address: string;
+    /** Either `"tcp"`, `"udp"` or `"unix"`. Defaults to `"tcp"`.
+     *
+     * In the future: `"tcp4"`, `"tcp6"`, `"udp4"`, `"udp6"`, `"ip"`, `"ip4"`,
+     * `"ip6"`, `"unix"`, `"unixgram"`, and `"unixpacket"`. */
+    transport: Transport;
```

Comment:
My typescript isn't so good. Can this just be `transport: "unix"` ?

(cc @kitsonk - for expert TS advice)

Reply:
There is quite a bit of typing cleanup that can be done around the listen() code - it is not great right now imo - I don't see why we even need the type Transport. I can address that in a later PR once this one is merged.

Reply:
Yeah, I think if we have overloads that express every "tagged" variant.  When I went through and cleaned up the comments I noticed this and though _ugh_.  I am not sure we need the seperate `Options` interfaces honestly.  There is a whole load of duplication that isn't necessary, and some tagged unions "should" work fine and actually should eliminate the overloads.

Reply:
@kitsonk I understand that the tagged unions can replace the overloads, but won't that force the return type to also be a union and therefore force if branching on the user side?